### DISCUSSION TITLE
About MCU overwrite strategy

### DISCUSSION THREAD

**Discussion by rouhento** [mcu-tools/mcuboot#1479] [General]
Hello,

I'm using MCU overwrite strategy and secondary slot (for upgrade) is located in external flash. The active FW is located right after the boot loader with 4kB header.

I have signed the FW upgrade image in the external flash memory with image tool this way:

imgtool sign --align 4 -v 1.2.4+4 --header-size 4096 --pad-header -S 0xE1000 fw_upraged_image.bin fw_upraged_image-signed.bin

Here is my configuration for the boot loader:

            "mcuboot.slot-size": "0xE1000",
            "mcuboot.max-img-sectors": "0x180",
            "mcuboot.read-granularity": 1,
            "mcuboot.signature-algorithm": "SIGNATURE_TYPE_NONE",
            "mcuboot.overwrite-only": true,
            "mcuboot.downgrade-prevention": true,
            "mcuboot.log-enable": true

I have copied upgrade FW image (version in bigger than the current FW) to the right place and mcuboot seems to recognise it because it fills the image slots with correct header information. But for some reason it does not copy the FW over the active one.

Any suggestions why? Which functions should be called that actually do the erasing of the primary slot and copying of the FW upgrade image in case of the overwrite-only type of upgrading?

**Comment by nvlsianpu** [comment]
You have to request upgrade for image collected in external slot.

**Reply by nvlsianpu** [reply]
ref. `int boot_set_pending(int permanent)`

**Reply by rouhento** [reply]
Thanks! That did it! Works now like a charm
